vpp/clientconfigsrv.go

```diff
@@ -0,0 +1,117 @@
+package vpp
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/satori/go.uuid"
+	"strings"
+)
+
+type ClientContext struct {
+	HostName string `json:"hostname"`
+	Guid     string `json:"guid"`
+}
+
+type Location struct {
+	LocationName string `json:"locationName"`
+	LocationId   int    `json:"locationId"`
+}
+
+type ClientConfigSrvResponse struct {
+	ClientContext      interface{} `json:"clientContext"`
+	AppleId            string      `json:"appleId"`
+	OrganizationIdHash string      `json:"organizationIdHash"`
+	Status             int         `json:"status"`
+	OrganizationId     int         `json:"organizationId"`
+	UId                string      `json:"uId"`
+	CountryCode        string      `json:"countryCode"`
+	Location           Location    `json:"location"`
+	ApnToken           string      `json:"apnToken"`
+	Email              string      `json:"email"`
+}
+
+type ClientConfigSrv struct {
+	ClientContext      ClientContext `json:"clientContext"`
+	AppleId            string        `json:"appleId"`
+	OrganizationIdHash string        `json:"organizationIdHash"`
+	Status             int           `json:"status"`
+	OrganizationId     int           `json:"organizationId"`
+	UId                string        `json:"uId"`
+	CountryCode        string        `json:"countryCode"`
+	Location           Location      `json:"location"`
+	ApnToken           string        `json:"apnToken"`
+	Email              string        `json:"email"`
+}
+
+func (c *Client) GetClientConfigSrv() (*ClientConfigSrv, error) {
+	request := map[string]interface{}{
+		"sToken": c.sToken,
+	}
+
+	clientConfigSrvUrl := c.VPPServiceConfigSrv.ClientConfigSrvUrl
+	var response ClientConfigSrvResponse
+	req, err := c.newRequest("POST", clientConfigSrvUrl, request)
+	if err != nil {
+		return nil, errors.Wrap(err, "create ClientConfigSrv request")
+	}
+
+	err = c.do(req, &response)
```

Comment:
if err := c.do...;err ! = nil { return } 
Always check errors

Reply:
This err is returned to be checked by the calling function.  I modeled it after "dep/device.go" lines 119 and 120. Is there something else I should be doing?